fuzzers/libafl-fuzz/src/main.rs

```diff
@@ -0,0 +1,246 @@
+#![feature(io_error_more)]
+#![deny(clippy::pedantic)]
+#![allow(clippy::unsafe_derive_deserialize)]
+#![allow(clippy::ptr_arg)]
+#![allow(clippy::unnecessary_wraps)]
+#![allow(clippy::module_name_repetitions)]
+#![allow(clippy::similar_names)]
+use std::{collections::HashMap, path::PathBuf};
+mod afl_stats;
+mod feedback;
+use clap::Parser;
+use corpus::{check_autoresume, remove_main_node_file};
+mod corpus;
+mod executor;
+mod fuzzer;
+mod hooks;
+mod utils;
+use fuzzer::run_client;
+use libafl::{
+    events::{CentralizedLauncher, EventConfig},
+    monitors::MultiMonitor,
+    Error,
+};
+use libafl_bolts::{
+    core_affinity::{CoreId, Cores},
+    shmem::{ShMemProvider, UnixShMemProvider},
+};
+use nix::sys::signal::Signal;
+use utils::PowerScheduleCustom;
+
+#[allow(clippy::too_many_lines)]
+fn main() {
+    env_logger::init();
+    let mut opt = Opt::parse();
+    executor::check_binary(&mut opt, SHMEM_ENV_VAR).expect("binary to be valid");
```

Comment:
Pass in the binary into the function instead of the `opt`, and give some useful output on failure

Reply:
``opt`` is necessary since there are several checks that may be skipped if frida, qemu, non instrumented, skip bin check etc

Reply:
Sad. Would still be good to pass in the needed flags only, but up to you. At least make the expect more meaningful?

Reply:
The errors returned from check_binary will also be printed when the expect is triggered so should be fine.